model/src/main/proto/exploration_checkpoint.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package model;
+
+import "exploration.proto";
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Top-level proto used to store exploration checkpoint per-profile basis.
+message ExplorationCheckpointDatabase {
+
+  // map from explorationId to ExplorationCheckpoint.
+  map<string, ExplorationCheckpoint> exploration_checkpoint = 1;
+}
+
+// Corresponds to the exploration checkpoint which will be used to resume the exploration.
+// This model contains information about all states of the deck which have been visited by the user.
+message ExplorationCheckpoint {
+
+  // The name of the latest state in the deck.
+  string pending_state_name = 1;
+
+  // The list of all states which the learner has previously completed.
+  repeated CompletedStateInCheckpoint completed_states_in_checkpoint = 2;
+
+  // Stores answers to the latest state in the deck.
+  repeated AnswerAndResponse pending_user_answers = 3;
+
+  // The Index of the current selected state from the deck of states.
```

Comment:
Nit: `The index ...` instead of `The Index`

Reply:
done